adapters/sparteo/sparteo.go

```diff
@@ -3,29 +3,41 @@ package sparteo
 import (
 	"fmt"
 	"net/http"
+	"net/url"
+	"strings"
+	"text/template"
 
 	"github.com/prebid/openrtb/v20/openrtb2"
 	"github.com/prebid/prebid-server/v3/adapters"
 	"github.com/prebid/prebid-server/v3/config"
 	"github.com/prebid/prebid-server/v3/errortypes"
+	"github.com/prebid/prebid-server/v3/macros"
 	"github.com/prebid/prebid-server/v3/openrtb_ext"
 	"github.com/prebid/prebid-server/v3/util/jsonutil"
+	"github.com/prebid/prebid-server/v3/util/ptrutil"
 )
 
 type adapter struct {
-	endpoint   string
-	bidderName string
+	endpoint *template.Template
 }
 
 type extBidWrapper struct {
 	Prebid openrtb_ext.ExtBidPrebid `json:"prebid"`
 }
 
+const unknownValue = "unknown"
+
 func Builder(bidderName openrtb_ext.BidderName, cfg config.Adapter, server config.Server) (adapters.Bidder, error) {
-	return &adapter{
-		endpoint:   cfg.Endpoint,
-		bidderName: string(bidderName),
-	}, nil
+	templateString := cfg.Endpoint + "?network_id={{.NetworkId}}{{if .SiteDomain}}&site_domain={{.SiteDomain}}{{end}}{{if .AppDomain}}&app_domain={{.AppDomain}}{{end}}{{if .Bundle}}&bundle={{.Bundle}}{{end}}"
```

Comment:
@t-sormonte 

Sorry if there was confusion here, we don't want conditionals in the template, either use the template without conditionals. Or if you don't want query params that are empty in your URL string, conditonally build the URL without using macros.